### ISSUE TITLE
Replacement of ReID model

### ISSUE THREAD

**Issue by TheMattBin** [yuanc3/Pose2ID#30]
Hi thanks for the great work!
I did a small experiment with other ReID model. I only compare the embedding after IPG+NFC towards my reid model in L2. It seems that the performance is quite bad.
Here, I didn't really change the feature extraction net during pose generation. It means that after pose generation, I do feature extraction with my own reid model and then with l2 normalization before NFC. 
I would like to know if it's possible to replace transreid with other reid model like this? Or I have to replace the whole reid_net during pose generation as well? Greatly appreciated if you can have some comments!

**Comment by yuanc3** [comment]
Please check there are l2 norms after fusion of IPG and NFC.

**Comment by TheMattBin** [comment]
Thanks for the prompted reply! Like the code you provided, after different pose generation, is it suitable to calculate cosine or l2 distance between feature embedding between reid and feats after IPG+NFC from your experience? Assuming they are all normalized.
```python
feats_ipg = torch.zeros_like(feats)
# fuse features of generated positive samples with different poses
for i in range(num_poses):
    feats_ipg += reid_model(feats_pose[i]) # <--- own model
eta = 1 # control the impact of generated images (considering the quality)
# centralize features and normalize to original distribution
feats = torch.nn.functional.normalize(feats + eta * feats_ipg, dim=1, p=2) # L2 normalization
from NFC import NFC
feats = NFC(feats)
```

**Comment by yuanc3** [comment]
Yes it’s fine. And after this code block (after NFC), add a l2 norm.

**Comment by TheMattBin** [comment]
Thanks a lot! Just curious why after NFC still needs normalization. Suppose I already have normalized and fused with reid model before NFC, does it mean that the features are already in normalized form? Also, would it be problematic to fused like below, I run reid model in batch processing and then with normalization and take mean as centralized feats:

```python
feats = F.normalize(feats, p=2, dim=1)
feats = torch.mean(feats, dim=0)
```

**Comment by yuanc3** [comment]
NFC may fuse different number of feats, so it needs a norm

And, the way you calculate the mean is ok.
